rules/qc.smk

```diff
@@ -0,0 +1,225 @@
+        """
+
+
+rule rseqc_junction_annotation:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        touch('star/rseqc/{sample}-{unit}.junction_annotation.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_junction_annotation/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_junction_annotation/{sample}-{unit}.stderr"
+    params:
+        extra = r'-q 255',  # STAR uses 255 as a scrore for uniq mappers
+        prefix= 'star/rseqc/{sample}-{unit}.junctionanno'
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        junction_annotation.py {params.extra} -i {input[0]} -r {input[1]} -o {params.prefix} > {log[0]} 2> {log[1]};
+        """
+
+        
+rule rseqc_junction_saturation:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        touch('star/rseqc/{sample}-{unit}.junction_saturation.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_junction_saturation/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_junction_saturation/{sample}-{unit}.stderr"
+    params:
+        extra = r'-q 255', 
+        prefix = 'star/rseqc/{sample}-{unit}.junctionsat'
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        junction_saturation.py {params.extra} -i {input[0]} -r {input[1]} -o {params.prefix} > {log[0]} 2> {log[1]};
+        """
+
+
+rule rseqc_stat:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+    output:
+        "star/rseqc/{sample}-{unit}.stats.txt"
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_stat/{sample}-{unit}.stderr"
+    params:
+        extra = r''
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        bam_stat.py {params.extra} -i {input} > {output} 2> {log}
+        """
+
+        
+rule rseqc_infer:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        "star/rseqc/{sample}-{unit}.infer_experiment.txt"
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_infer/{sample}-{unit}.stderr"
+    params:
+        extra = r''
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        infer_experiment.py {params.extra} -r {input[1]} -i {input[0]} > {output} 2> {log}
+        """
+
+        
+rule rseqc_genebody:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        touch("star/rseqc/{sample}-{unit}.genebody.done")
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_genebody/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_genebody/{sample}-{unit}.stderr"
+    params:
+        extra = r'',
+        prefix = "star/rseqc/{sample}-{unit}.geneBodyCoverage"
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        geneBody_coverage.py {params.extra} -r {input[1]} -i {input[0]} -o {params.prefix} > {log[0]} 2> {log[1]}
+        """
```

Comment:
When having simple commands it is more concise without hampering readability to write them like this:
```
    shell:
        "geneBody_coverage.py {params.extra} -r {input[1]} -i {input[0]} -o {params.prefix} "
        "> {log[0]} 2> {log[1]}"
```

Python string concatenation magic.

Reply:
Also, usually it is fine to just have one log. Is there a specific need for splitting up into stdout and stderr?
Second, when referring to multiple files, do it by name for increased readability.
```
rule rseqc_genebody:
    input:
        bam="star/{sample}-{unit}/Aligned.out.bam",
        bed="star/rseqc/annotation.bed"
    output:
        touch("star/rseqc/{sample}-{unit}.genebody.done")
    priority: 1
    log:
        "logs/rseqc/rseqc_genebody/{sample}-{unit}.log"
    params:
        prefix = "star/rseqc/{sample}-{unit}.geneBodyCoverage"
    conda:
        "../envs/rseqc.yaml"
    shell:
        "geneBody_coverage.py {params.extra} -r {input.bed} -i {input.bam} -o {params.prefix} "
        "> {log} 2>&1"
```

Finally, I'd rather not use touch in all these rules and instead list the specific output files. Any reason not to do this?

Reply:
ok